### DISCUSSION TITLE
Mod minecraft 1.16.5 fabric First Person Mod

### DISCUSSION THREAD

**Discussion by Ak2mE** [FabricMC/fabric-api#3273] [Mod Dev Support]
Please help me I do not know how to solve the problem. 

thank you in advance ;)

---- Minecraft Crash Report ----
// I blame Dinnerbone.

Time: 20.08.2023, 10:47
Description: Unexpected error

java.lang.NullPointerException: Cannot invoke "dev.tr7zw.transliterationlib.api.wrapper.util.Keybind.isPressed()" because "dev.tr7zw.firstperson.FirstPersonModelCore.keyBinding" is null
	at Not Enough Crashes deobfuscated stack trace.(1.16.5+build.10)
	at dev.tr7zw.firstperson.FirstPersonModelCore.onTick(FirstPersonModelCore.java:61)
	at dev.tr7zw.firstperson.fabric.FirstPersonModelMod.lambda$onInitialize$0(FirstPersonModelMod.java:27)
	at net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents.lambda$static$2(ClientTickEvents.java:65)
	at net.minecraft.client.MinecraftClient.handler$zjb000$fabric-lifecycle-events-v1$onEndTick(MinecraftClient:5386)
	at net.minecraft.client.MinecraftClient.tick(MinecraftClient:1612)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1021)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:681)
	at net.minecraft.client.main.Main.main(Main:215)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 1720980664 bytes (1641 MB) / 2147483648 bytes (2048 MB) up to 3221225472 bytes (3072 MB)
	CPUs: 2
	JVM Flags: 13 total; -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:+ParallelRefProcEnabled -Xms2048M -Xmx3072M -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss2M
	Suspected Mods: FirstpersonModel (firstperson), Fabric Loader (fabricloader)
	Fabric Mods: 
		advanced_runtime_resource_pack: Runtime Resource Pack 0.5.1
		alternate-current: Alternate Current 1.5.0
		autoconfig1u: Auto Config v1 Updated 3.3.1
		betterbeds: Better Beds 1.1.0
		betterbiomeblend: Better Biome Blend 1.16.4-1.2.9-fabric
		c2me: Concurrent Chunk Management Engine 0.1-SNAPSHOT
		caffeineconfig: CaffeineConfig 1.0.0
		clientsidenoteblocks: Clientsidenoteblocks 2.0
		cloth-basic-math: cloth-basic-math 0.6.1
		cloth-config2: Cloth Config v4 4.16.91
		collective-fabric: Collective (Fabric) 5.50
		com_moandjiezana_toml_toml4j: toml4j 0.7.2
		cullleaves: Cull Leaves 2.1.0
		dynamicfps: Dynamic FPS 2.0.2
		enhancedblockentities: Enhanced Block Entities 0.4
		entityculling: EntityCulling-Fabric 1.5.2
		extrasounds: Extra Sounds 2.0.1+1.16
		fabric: Fabric API 0.42.0+1.16
		fabric-api-base: Fabric API Base 0.4.0+3cc0f0907d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.3.1+3cc0f0907d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.13+3cc0f0907d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.6+3cc0f0907d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.3+3cc0f0907d
		fabric-commands-v0: Fabric Commands (v0) 0.2.3+3cc0f0907d
		fabric-containers-v0: Fabric Containers (v0) 0.1.12+3cc0f0907d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.5+3cc0f0907d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.4+3cc0f0907d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.8+3cc0f0907d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.2.4+3cc0f0907d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.5+3cc0f0907d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.2+3cc0f0907d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.7+3cc0f0907d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.2+3cc0f0907d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.1+3cc0f0907d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.5+3cc0f0907d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.2+3cc0f0907d
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.2+3cc0f0907d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.3+3cc0f0907d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.4+3cc0f0907d
		fabric-models-v0: Fabric Models (v0) 0.3.1+3cc0f0907d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.5+3cc0f0907d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.9+3cc0f0907d
		fabric-networking-v0: Fabric Networking (v0) 0.3.3+3cc0f0907d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.6+3cc0f0907d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.3+3cc0f0907d
		fabric-particles-v1: Fabric Particles (v1) 0.2.5+3cc0f0907d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.6+3cc0f0907d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.5+3cc0f0907d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.5+3cc0f0907d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.3.1+3cc0f0907d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.6+3cc0f0907d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.15+3cc0f0907d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.3+3cc0f0907d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.1+3cc0f0907d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.8+3cc0f0907d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.1+3cc0f0907d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.6+3cc0f0907d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.12+3cc0f0907d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.2+3cc0f0907d
		fabric-textures-v0: Fabric Textures (v0) 1.0.7+3cc0f0907d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.8+3cc0f0907d
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.0+3cc0f0907d
		fabricloader: Fabric Loader 0.14.18
		fastchest: FastChest 1.2+1.16
		ferritecore: FerriteCore 2.1.1
		fiber: fiber 0.23.0-2
		firstperson: FirstpersonModel 2.0.0
		freecam: Freecam 1.2.1+1.16
		indium: Indium 1.0.0+mc1.16.5
		invmove: InvMove 0.8.4
		itemmodelfix: Item Model Fix 1.0.3+1.16
		jankson: Jankson 3.0.1+j1.2.0
		java: OpenJDK 64-Bit Server VM 16
		kirin: Kirin UI 1.7-1.16.2
		krypton: Krypton 0.1.2
		lazydfu: LazyDFU 0.1.1
		libgui: LibGui 3.4.0+1.16.5
		lithium: Lithium 0.6.6
		logical_zoom: Logical Zoom 0.0.8
		minecraft: Minecraft 1.16.5
		modernfix: ModernFix 5.6.1+mc1.16.5
		modmenu: Mod Menu 1.16.23
		notenoughanimations: NotEnoughAnimations 1.2.4
		notenoughcrashes: Not Enough Crashes 4.1.4+1.16.5
		overworld_two: Overworld 2 1.0.4
		presencefootsteps: Presence Footsteps r29+1.16.4
		reeses-sodium-options: Reese's Sodium Options 1.5.0+mc1.16.5-build.68
		smoothboot: Smooth Boot 1.16.5-1.5.0
		sodium: Sodium 0.2.0+build.4
		sodium-extra: Sodium Extra 0.4.5+mc1.16.5-build.40
		starlight: Starlight fabric-1.0.0-RC2-82ae5ca
		transliterationlib: TRansliterationLib 1.0.4
		wthit: wthit 2.10.15
	Launched Version: Fabric 1.16.5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: GeForce 610M/PCIe/SSE2 GL version 4.6.0 NVIDIA 388.71, NVIDIA Corporation
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: vanilla, Fabric Mods, file/Colorful+Ping+Indicator+6.zip, file/Water+Improved.zip, file/Soft-Weather-1.0-1.16.zip, file/LowOnFire+1.16+.zip, file/Goshas_RPG_v2.1.zip, file/EBE-Models-OptimizedPack_v1.2_mc1.16-1.20.zip (incompatible), file/Smaller Totem.zip (incompatible), file/UniqueDark_1.16.5.zip
	Current Language: English (US)
	CPU: 2x Intel(R) Pentium(R) CPU B970 @ 2.30GHz
	Client Crashes Since Restart: 1
	Integrated Server Crashes Since Restart: 0

Download file - [crash-2023-08-20_10.47.56-client.txt](https://github.com/FabricMC/fabric/files/12387776/crash-2023-08-20_10.47.56-client.txt)

**Comment by warjort** [comment]
java.lang.NullPointerException: Cannot invoke "dev.tr7zw.transliterationlib.api.wrapper.util.Keybind.isPressed()" because "dev.tr7zw.firstperson.FirstPersonModelCore.keyBinding" is null
at Not Enough Crashes deobfuscated stack trace.(1.16.5+build.10)
at dev.tr7zw.firstperson.

> FirstPersonModel

Core.onTick(FirstPersonModelCore.java:61)

Issue with the FirstPersonModel mod, check you have the latest version then contact the mod author

**Reply by Ak2mE** [reply]
Thank you
